### PR TITLE
[FINAL] Modify MOAI to 64 bit

### PR DESCRIPTION
**Pull request by tonidero**

### CODE REVIEW

File: xcode/libmoai/libmoai.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -14751,7 +14752,8 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = B315CD8C1806128E00F617D4 /* libmoai-ios.xcconfig */;
 			buildSettings = {
-				"ARCHS[sdk=iphoneos*]" = "$(ARCHS_STANDARD_32_BIT)";
+				ARCHS = "$(ARCHS_STANDARD_INCLUDING_64_BIT)";
+				"ARCHS[sdk=iphoneos*]" = "$(ARCHS_STANDARD_INCLUDING_64_BIT)";
```
Comment (NachoSoto) [inline xcode/libmoai/libmoai.xcodeproj/project.pbxproj:14756]: We should probably put this in the `xcconfig` files. Let me know if you want me to show you what they are :) (if you don't already know).

**Comment by jeiting** [comment]
I can't read the diff but a third of a million lines removed can't be bad.

**Comment by tonidero** [comment]
Yeah it mainly has to do with the removed libs. Some of them had a lot of code

File: 3rdparty/curl-7.19.7/include-iphone/curl/curlrules.h [hunk 2]
```diff
@@ -130,7 +130,8 @@
  * Macros private to this header file.
  */
 
-#define CurlchkszEQ(t, s) sizeof(t) == s ? 1 : -1
+//#define CurlchkszEQ(t, s) sizeof(t) == s ? 1 : -1
```
Comment (NachoSoto) [inline 3rdparty/curl-7.19.7/include-iphone/curl/curlrules.h]: This is commented out? Why the change to just `1`?

File: 3rdparty/curl-7.19.7/lib/url.c [hunk 3]
```diff
@@ -763,19 +763,6 @@ CURLcode Curl_open(struct SessionHandle **curl)
 
   data->magic = CURLEASY_MAGIC_NUMBER;
 
-#ifdef USE_ARES
```
Comment (NachoSoto) [inline 3rdparty/curl-7.19.7/lib/url.c:766]: Do we need `curl` anyway?

File: 3rdparty/lua-5.1.3/src/lauxlib.c [hunk 4]
```diff
@@ -186,7 +186,7 @@ LUALIB_API lua_Number luaL_optnumber (lua_State *L, int narg, lua_Number def) {
 }
 
 
-LUALIB_API lua_Integer luaL_checkinteger (lua_State *L, int narg) {
+LUALIB_API int luaL_checkinteger (lua_State *L, int narg) {
```
Comment (NachoSoto) [inline 3rdparty/lua-5.1.3/src/lauxlib.c]: Shouldn't we use the typedef here?

File: 3rdparty/curl-7.19.7/include-iphone/curl/curlrules.h (hunk 2 above)
Comment (tonidero) [inline 3rdparty/curl-7.19.7/include-iphone/curl/curlrules.h]: Not sure why but this operation seems to return -1 when it's used a few lines later (145) and produces an error for creating an array with a negative size

File: 3rdparty/curl-7.19.7/lib/url.c (hunk 3 above)
Comment (tonidero) [inline 3rdparty/curl-7.19.7/lib/url.c:766]: That's one of the libraries I tried to remove but it seems that there are some methods used in CoreMS so I preferred to leave it

Comment (NachoSoto) [inline 3rdparty/curl-7.19.7/lib/url.c:766]: Do you remember which ones?

File: 3rdparty/lua-5.1.3/src/lauxlib.c (hunk 4 above)
Comment (tonidero) [inline 3rdparty/lua-5.1.3/src/lauxlib.c]: Not sure why but lua_Integer is long. However I will change this back and do a cast where the method is used which seems better

Comment (NachoSoto) [inline 3rdparty/lua-5.1.3/src/lauxlib.c]: Right, because Integers are 64 bits in a 64 bit platform :)

**Comment by jeiting** [comment]
If achilles is running the games with the fork I'm confident that this works.

:ship: :it:

**Comment by NachoSoto** [comment]
:shipit: after we fix that comment out line :)

**Comment by NachoSoto** [comment]
Merging in 5 minutes!
4:59
4:58
...

**Comment by jeiting** [comment]
I like this. Merge timer.

**Comment by NachoSoto** [comment]
I'm actually waiting to see what happens with #37.

**Comment by jeiting** [comment]
I wouldn't worry about that. Just a bad PR.

**Comment by jeiting** [comment]
- 622 additions
- 447,419 deletions

**Comment by NachoSoto** [comment]
:boom: Nice job @tonidero!

**Comment by tonycosentini** [comment]
Does it still build for Android?
-- 
Tony Cosentini

On November 20, 2013 at 4:28:10 PM, NachoSoto (notifications@github.com) wrote:

 Nice job @tonidero!

**Comment by jeiting** [comment]
Yeah, I was kind of wondering that. Probably not considering all the things that were removed.

**Comment by NachoSoto** [comment]
Oh shit we should have checked that.

**Comment by jeiting** [comment]
cross platform problems

**Comment by tonidero** [comment]
Yeah, didn't thought about that... I'm not sure but most of the things removed are from 3rd party libs not sure which ones are used in android though...

**Comment by tonycosentini** [comment]
I just thought of it :-\

Should we merge this and then start a new branch for Android support?
-- 
Tony Cosentini

On November 20, 2013 at 4:30:41 PM, Jacob Eiting (notifications@github.com) wrote:

cross platform problems

**Comment by jeiting** [comment]
Yeah.

**Comment by jeiting** [comment]
Just need to update the .mk files i think.

**Comment by tonycosentini** [comment]
Yeah, it'll be mostly updating mk files.

The Android build system for this is really bizarre though, I can give you an overview @tonidero, shouldn't be _too_ bad.

**Comment by tonycosentini** [comment]
Also, the diff for this is amazing.

**Comment by tonidero** [comment]
Yeah sure! :) And well, finally more than a 64bit branch it became a cleaning branch :laughing:
